cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -468,11 +468,13 @@ func GetAvailableResources(clientBuilder clientbuilder.ControllerClientBuilder)
 // the shared-informers client and token controller.
 func CreateControllerContext(s *config.CompletedConfig, rootClientBuilder, clientBuilder clientbuilder.ControllerClientBuilder, stop <-chan struct{}) (ControllerContext, error) {
 	versionedClient := rootClientBuilder.ClientOrDie("shared-informers")
+
+	// always stop informers when the ListAndWatch errors out
+	onListErr := func(error) bool { return true }
 	sharedInformers := informers.NewSharedInformerFactory(versionedClient, ResyncPeriod(s)())
 
 	metadataClient := metadata.NewForConfigOrDie(rootClientBuilder.ConfigOrDie("metadata-informers"))
-	metadataInformers := metadatainformer.NewSharedInformerFactory(metadataClient, ResyncPeriod(s)())
-
+	metadataInformers := metadatainformer.NewSharedInformerFactoryWithOptions(metadataClient, ResyncPeriod(s)(), metadatainformer.WithOnListError(onListErr))
```

Comment:
fix `ResyncPeriod` so that it's just `ResyncPeriod(s)` not `ResyncPeriod(s)()`

Reply:
(might as well so that we're consistent)

Reply:
yeah, better to be consistent, i think

Reply:
Upon second look, I don't think this is inconsistent like we thought (or at least how I thought).

I had thought that `defaultResync` was a `SharedInformerOption` that was passed as part of the multivariate options argument to the factory. It's actually a required argument that is just a duration. The reason it looks funky here (by returning a function that is called) is because it is supposed to generate a random duration anytime it gets invoked.

It's a little weird, but I don't think it's inconsistent (because it's not actually an option) and I'm not sure there's a way to change it without breaking existing interfaces. Let me know if you still think we should change it.

Reply:
After speaking with Solly, we've agreed to leave it as is, I've left a not on `ResyncPeriod` to clarify why it returns a function instead of a duration.